### ISSUE TITLE
Enable full-size (normal width) scrollbar

### ISSUE THREAD

**Issue by grahamperrin** [piroor/treestyletab#2489]
I see no scrollbar: 

![image](https://user-images.githubusercontent.com/192271/75096603-28239800-5599-11ea-8908-5dd95e313f87.png)

– and with a trackball device for input, I can't scroll. As a workaround I typically use keyboard shortcuts to navigate tabs, or switch temporarily to Sidebery. 

I see e.g. preferences for `scrollbarMode` and `sidebarScrollbarPosition` but I can't guess the effects of changing the values in version 3.3.6. 

Shot of a cleaner profile: 

![image](https://user-images.githubusercontent.com/192271/75096505-0a096800-5598-11ea-9186-0a3fced14a55.png)

Can I force appearance of the scrollbar? 

(If I'm overlooking something, I'm sincerely sorry; I spent half an hour or so searching including e.g. the tabs in the first screenshot.)

```text
grahamperrin@momh167-gjp4-8570p:~ % date ; uname -v
Sat 22 Feb 2020 17:24:42 GMT
FreeBSD 13.0-CURRENT #51 r357746: Tue Feb 11 03:54:19 GMT 2020     root@momh167-gjp4-8570p:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG 
grahamperrin@momh167-gjp4-8570p:~ % pkg query '%o %v %R' firefox
www/firefox 73.0.1,1 FreeBSD
grahamperrin@momh167-gjp4-8570p:~ % 
```

TIA

**Comment by irvinm** [comment]
I see the scrollbar in both screenshots. (I think)  It looks like it is on the left side of the sidebar and it is "half-sized".  You can make it full-sized if you want ... or even move it to the right side of the sidebar if it is easier to see and\or interact with your trackball.  You can also change the color scheme of the scrollbar to make it easier to see.

It looks like in the 2nd screenshot, the scrollbar is all the way to the bottom but I am not sure why the last tab gets clipped by the "Open a new tab" "+" bar ...

**Comment by grahamperrin** [comment]
Oh, wow! Thanks. 

Now I see it, in the screenshots, but the half-size is (for me) very vague. The previous screenshots were slightly misleading in that the shot software excluded the borders. Here's a truer representation: 

![image](https://user-images.githubusercontent.com/192271/75102556-4c0ccb00-55e5-11ea-9746-bff05fb0acb3.png)

– at a glance, the bar is almost as narrow as the indicators alongside each task, almost as narrow as borders. 

----

Now, embarrassingly, I can't find an option to widen the scrollbar. 

From <https://github.com/piroor/treestyletab/issues/1544#issuecomment-522902490>: 

> … "Preferences" > Scroll all the way down to an expandable menu named "All Configs" …

A few hours ago this expandable menu was visible to me. Now it's not: 

![image](https://user-images.githubusercontent.com/192271/75102668-2c76a200-55e7-11ea-85a0-17502f0f83f8.png)

I considered the possibility of an extensions conflict but with a different profile – Tree Style Tab alone enabled – here, too, I can't see _All Configs_: 

![image](https://user-images.githubusercontent.com/192271/75102713-f4bc2a00-55e7-11ea-9221-1a7f55eaa39c.png)

Have I become word-blind? Or am I confusing myself in some other way?

**Comment by grahamperrin** [comment]
> I can't see _All Configs_:

OK, now I see it, expandable _All Configs_ nested under expandable _Development_ but still (sorry) I can't find anything to normalise the width of the scrollbar. 

It's past 03:00 in the morning here, entirely possible that I'm doing something stupid (again) …

**Comment by irvinm** [comment]
You can use CSS to take care of it:  https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#show-scrollbar-with-regular-width

**Comment by grahamperrin** [comment]
Yay! 

Five stars, https://addons.mozilla.org/addon/tree-style-tab/reviews/1494551/

![image](https://user-images.githubusercontent.com/192271/75112824-9a4eb800-563f-11ea-9787-272f6ac2001c.png)

* `#tabbar { scrollbar-width: auto; }`
* Vertigo appearance
* style of contents for the sidebar position: right side (I used Macs for around twenty years, I still prefer close buttons to the left)
* crop with ".."

**Comment by mxmilkb** [comment]
I have tried this in userChrome.css![image](https://user-images.githubusercontent.com/108225/81416426-9eb84680-9141-11ea-911a-bc3740067729.png) and the textarea in Advanced / Extra Style Rules... but I'm still getting the narrower scrollbar. Might there be something else in the options overriding that style?

**Comment by irvinm** [comment]
@mxmilkb you only need to add "#tabbar { scrollbar-width: auto; }" to your TST CSS.  I tried your 20px value and it didn't work for me either.  I am guessing that it not a valid size for the scrollbar.

**Comment by piroor** [comment]
* [scrollbar-width - CSS: Cascading Style Sheets | MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-width)
* [CSS Scrollbars Module Level 1](https://drafts.csswg.org/css-scrollbars-1/#scrollbar-width)

The description at MDN says that the `scrollbar-width` property accepts CSS length, but the CSS spec actually says that it accepts only three keywords `none`, `auto`, and `thin`. Indeed Firefox says that length like `40px` is invalid for the property (you can see the error on the DOM inspector).

**Comment by mxmilkb** [comment]
Thanks folks, apologies, I had thought I had tried "auto" before with no affect, but I might have typoed or something as it does work.

**Comment by zesty** [comment]
workaround is no longer working

**Comment by piroor** [comment]
`#tabbar { scrollbar-width: auto; }` still works for me on Nightly 102.0a1 (on Windows 10). Screenshot:

![image](https://user-images.githubusercontent.com/70062/167346626-7a228999-7ada-4ab7-9b7b-92eb5914a040.png)

**Comment by zesty** [comment]
more info:

I'm on Kubuntu 22.04 / ff 100.0

The scollbar gutter appears and disappears -- with mouseover -- as an overlay, taking up no structural/layout width, such that it covers most of the click area for expand/collapsing trees when active (scrollbar on left side), making for unexpected scrolling when trying to expand/collapse. Before ff 100, the scrollbar gutter took up real/layout space all the time. Now the gutter is present only when mouse-hovered (no animation).

Changing up the themes (addon and firefox) didn't help, neither did L/R positioning (scrollbar or sidebar), nor disable/enable the addon.

Scrollbar slider not shown:

![1](https://user-images.githubusercontent.com/262676/167387547-4f373f1e-67ad-47a9-8dbf-afe4bdad4e54.png)
![2](https://user-images.githubusercontent.com/262676/167387575-c9eb42b3-f2c7-4ea3-ad31-2e7b5c5d11c6.png)

**Comment by irvinm** [comment]
@zesty have you tried removing all of your other TST CSS and just add the one line back for the scrollbar back in to see if that works?  Also, do you have anything in your userChrome.css?  Maybe try disabling that for testing purposes too.

I tried on Ubuntu 22.04 and didn't see any problems.

**Comment by zesty** [comment]
TL;DR workaround: enable double click tab title to expand collapse and use that instead

Preferences > Tree Behavior > Double-click on a tab > Collapse/expand tree

---

Don't have a userChrome.css; no joy with the clean TST CSS.  Seems like it might be something to do with kde/qt/plasma interaction with ff:

[kde] [Bug 439448] New: Scrollbars are invisible and disappear, and are too narrow
https://www.mail-archive.com/kde-bugs-dist@kde.org/msg582213.html

439448 – Scrollbars are invisible and disappear, and are too narrow
https://bugs.kde.org/show_bug.cgi?id=439448

I hammered on the kde/plasma appearance settings, and tried some of the standard options in qt5ct to no effect. I'll update again if I can find a fix, and maybe check back to see if anyone else reports. ¯\_(ツ)_/¯

**Comment by piroor** [comment]
Is this workaround effective on this case?: https://github.com/piroor/treestyletab/discussions/3124#discussioncomment-2742198

**Comment by zesty** [comment]
Yep, that was indeed the issue.
